### PR TITLE
mongosh: init at 1.5.0

### LINKED ISSUE #159419: Request to add mongosh package
**Project description**
The MongoDB Shell

**Metadata**

* homepage URL: https://github.com/mongodb-js/mongosh
* source URL: https://github.com/mongodb-js/mongosh
* license: APACHE LICENSE, VERSION 2.0
* platforms: linux, darwin

### PR DESCRIPTION
**Pull request by aaronjheng**
###### Motivation for this change

The MongoDB Shell, mongosh, is a fully functional JavaScript and Node.js 14.x REPL environment for interacting with MongoDB deployments. You can use the MongoDB Shell to test queries and operations directly with your database.

Resolves  #159419

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/mongosh/generate.sh [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+rm -rf gen && mkdir -p gen
+
+node2nix \
+    --no-copy-node-env \
+    --node-env ../../node-packages/node-env.nix \
+    --input packages.json \
+    --output gen/packages.nix \
+    --composition gen/composition.nix \
+    --strip-optional-dependencies \
+    --nodejs-16
```
Comment (SuperSandro2000) [inline pkgs/development/tools/mongosh/generate.sh:19]: This does not match top-level

**Review by SuperSandro2000 — COMMENTED** [review]
Those are all the default files. Why not add this to nodePackages?

**Comment by aaronjheng** [comment]
Mongosh is kinda slow in support for newer node versions. I think keeping it separate from nodePackages would be good for maintenance.

**Comment by aaronjheng** [comment]
@SuperSandro2000 Should we downgrade mongosh or mark it broken?

**Comment by SuperSandro2000** [comment]
> @aaronjheng naively I wonder why not just update `mongosh` to use `nodejs-16_x`?

Yeah, I don't see a blocker why we couldn't do this.

**Comment by aaronjheng** [comment]
Actually, there is a blocking issue. See https://github.com/NixOS/nixpkgs/issues/132456.

**Pull request merged**